### PR TITLE
Update lora def

### PR DESCRIPTION
**Pull request by lucylq**
### Summary
Update lora def to use nn.linear instead of torch.nn.functional.linear

1. Remove debug string from FQN. Otherwise we get different quant indices for lora and non-lora (because lora_a is quantized), after this change, and the foundation weights are no longer shareable.
2. Having nn.Linear as a submodule allows torchao quant to capture it, and we don't need custom logic filtering LoraLinear. lora_a and lora_b are also captured, which changes our results slightly. Note that only lora_a is quantized (shape [16, dim], where dim %32 ==0), as lora_b has shape [dim, 16], where 16 %32 != 0 and fails the group size check. 
3. Having nn.Linear as submodule also means we need to remap lora weights, as weight names have an extra 'linear' in them.
4. Add @property for weight, bias so that it remains BC and can be treated as a regular linear module. This is used in [load_weights_from_attention.py](https://github.com/pytorch/executorch/blob/eb92ceca74d0bace6d01115ae7f860b3baed5c2c/examples/models/llama/static_attention.py#L1154

### Test plan
CI

### CODE REVIEW

File: examples/models/llama/lora.py [hunk 1]
```diff
@@ -28,20 +28,25 @@ def __init__(
...
+        # Remap old-style "weight" key to "linear.weight" for backward compat
+        old_key = prefix + "weight"
+        new_key = prefix + "linear.weight"
+        if old_key in state_dict and new_key not in state_dict:
+            state_dict[new_key] = state_dict.pop(old_key)
+        super()._load_from_state_dict(state_dict, prefix, *args, **kwargs)
+
     def forward(self, x: torch.Tensor) -> torch.Tensor:
-        out = torch.nn.functional.linear(x, self.weight, self.bias)
+        out = self.linear(x)
```
[the remark this answers is not in the document]
Comment (lucylq) [reply examples/models/llama/lora.py:53]: added bias property

File: backends/xnnpack/operators/node_visitor.py [hunk 2]
```diff
@@ -625,7 +625,7 @@ def get_serialized_buffer_index(
             f"Serializing constant data node {tensor} but tensor value has no bytes",
         )
         sha256_hash = hashlib.sha256(bytes(array))
-        named_key = tensor.name + "_" + sha256_hash.hexdigest()
+        named_key = sha256_hash.hexdigest()
```
[the remark this answers is not in the document]
Comment (lucylq) [reply backends/xnnpack/operators/node_visitor.py:629]: I think this is fine, we should error out if there's duplicate name in ptd, pte files. I added the tensor name for debuggability earlier, but seems like this is problematic with the generated quant names.

**Comment by lucylq** [comment]
Ah, I will add a config to disable/enable lora quant.

**Comment by lucylq** [comment]
move to here: https://github.com/pytorch/executorch/pull/18342
